File: services/gateway/app/models/login_attempt.py
import uuid
from datetime import datetime, timezone

from sqlalchemy import Boolean, Column, DateTime, String
from sqlalchemy.dialects.postgresql import UUID

from app.db.session import Base


def _now() -> datetime:
    return datetime.now(timezone.utc)


class LoginAttempt(Base):
    __tablename__ = "login_attempts"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    email = Column(String(180), nullable=False, index=True)
    success = Column(Boolean, nullable=False)
    remote_ip = Column(String(64), nullable=True)
    attempted_at = Column(DateTime(timezone=True), default=_now, nullable=False, index=True)

File: services/core/src/main/java/com/previopls/entity/LoginAttempt.java
package com.previopls.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.Instant;
import java.util.UUID;

@Entity
@Table(name = "login_attempts", indexes = {
        @Index(name = "ix_login_attempts_email", columnList = "email"),
        @Index(name = "ix_login_attempts_attempted_at", columnList = "attempted_at"),
        @Index(name = "ix_login_attempts_email_attempted_at", columnList = "email, attempted_at")
})
@Getter
@Setter
@NoArgsConstructor
public class LoginAttempt {

    @Id
    @GeneratedValue
    @Column(columnDefinition = "uuid")
    private UUID id;

    @Column(nullable = false, length = 180)
    private String email;

    @Column(nullable = false)
    private boolean success;

    @Column(name = "remote_ip", length = 64)
    private String remoteIp;

    @Column(name = "attempted_at", nullable = false, updatable = false)
    private Instant attemptedAt = Instant.now();
}
